addons/mail/models/mail_thread.py

```diff
@@ -2977,9 +2977,15 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
             } for pid in inbox_pids]
             self.env['mail.notification'].sudo().create(notif_create_values)
 
-            message_format_values = message.message_format(msg_vals=msg_vals)[0]
+            MailMessage = self.env['mail.message']
+            messages_format_prepared = MailMessage._message_format_personalized_prepare(
+                message.message_format(msg_vals=msg_vals), partner_ids=inbox_pids)
             for partner_id in inbox_pids:
-                bus_notifications.append((self.env['res.partner'].browse(partner_id), 'mail.message/inbox', dict(message_format_values)))
+                bus_notifications.append(
+                    (self.env['res.partner'].browse(partner_id),
+                     'mail.message/inbox',
+                     dict(MailMessage._message_format_personalize(partner_id, messages_format_prepared)[0]))
```

Comment:
Why forcing the dict ? Isn't it already a list of dict, hence a dict when taking the first item ?